sailthru/Sailthru_Client.php

```diff
@@ -1434,9 +1434,22 @@ protected function httpRequestCurl($action, array $data, $method = 'POST', $opti
         curl_setopt($ch, CURLOPT_HTTPHEADER, $this->httpHeaders);
         $response = curl_exec($ch);
         $this->lastResponseInfo = curl_getinfo($ch);
+
+        // Get the curl error message if there is one
+        $errorMessage = false === $response ? curl_error($ch) : null;
+
         curl_close($ch);
 
+        if (false === $response) {
+            // There's a curl error! throw an exception which gives us some details
+            throw new Sailthru_Client_Exception(
+                sprintf('Error with curl transport: %s', $errorMessage),
+                Sailthru_Client_Exception::CODE_TRANSPORT_ERROR
+            );
+        }
+
         if (!$response) {
+            // We have an empty (well, falsey) response for the server, but not a curl error
```

Comment:
If you mock up an HTTP server, it seems to throw for me.

e.g. 
```bash
$ http-server
> running on port 8080
$ php.. getUser('test@sailthru.com', []);
<Sailthru_Client_Exception::Empty_Response>
```

Also - {$response} is empty when if $response is false, so a placeholder or alternative message should at the least be applied there.